middleware/http/source/inbound/call.cpp

```diff
@@ -0,0 +1,363 @@
+               result.payload.header.emplace_back( "content-type", http::protocol::convert::from::buffer( message.buffer.type));
+               result.payload.header.emplace_back( http::header::name::result::code, http::header::value::result::code( message.code.result));
+               result.payload.header.emplace_back( http::header::name::result::user::code, http::header::value::result::user::code( message.code.user));
+
+               result.payload.body = std::move( message.buffer.data);
+
+               auto transform_http_code = []( auto code)
+               {
+                  switch( code)
+                  {
+                     case common::code::xatmi::ok: return http::code::ok;
+                     case common::code::xatmi::timeout: return http::code::request_timeout;
+                     // TODO what to return? can't be a 200. 207?
+                     case common::code::xatmi::service_fail: return http::code::internal_server_error;
+
+                     default:
+                        return http::code::internal_server_error;
+                  }
+               };
+
+               result.code = transform_http_code( message.code.result);
+               
+               return result;
+            }
+            */
+
+            namespace send
+            {
+               auto lookup( const communication::ipc::inbound::Device& ipc, const std::string& service)
+               {
+                  message::service::lookup::Request lookup{ local::handle( ipc)};
+                  lookup.requested = service;
+                  lookup.context.semantic = decltype( lookup.context.semantic)::no_busy_intermediate;
+                  return communication::device::blocking::send( communication::instance::outbound::service::manager::device(), lookup);
+               }
+
+               auto request( const communication::ipc::inbound::Device& ipc, message::service::lookup::Reply lookup, message::service::call::callee::Request request)
+               {
+                  if( lookup.state == decltype( lookup.state)::absent)
+                     common::code::raise::error( common::code::xatmi::no_entry, "failed to lookup service: ", lookup.service.name);
+
+                  request.process = local::handle( ipc);
+                  request.service = lookup.service;
+                  request.pending = lookup.pending;
+                  return communication::device::blocking::send( lookup.process.ipc, request);
+               }
+            } // send
+            
+            template< typename Policy>
+            struct basic_caller
+            {
+               basic_caller( communication::ipc::inbound::Device ipc, Request request)
+                  : correlation{ send::lookup( ipc, request.service)}, ipc{ std::move( ipc)}, request{ Policy::transform( std::move( request))}
+               {}
+
+               ~basic_caller()
```

Comment:
`~basic_caller() = default;`

Or leave it out, and let the compiler generate one?